### ISSUE TITLE
uploading image

### ISSUE THREAD

**Issue by margOnline** [FanaHOVA/dfs-article#1]
There is an error from imgur-api gem:
```
ruby/gems/2.2.0/gems/imgur-api-0.0.4/lib/imgur.rb:244:in `at': can't convert nil into an exact number (TypeError)
```

The full stack trace is
```
ruby/gems/2.2.0/gems/imgur-api-0.0.4/lib/imgur.rb:244:in `initialize'
ruby/gems/2.2.0/gems/imgur-api-0.0.4/lib/imgur.rb:71:in `new'
ruby/gems/2.2.0/gems/imgur-api-0.0.4/lib/imgur.rb:71:in `upload'
twilio-dfs-tracker/notification.rb:47:in `upload_image'
twilio-dfs-tracker/notification.rb:26:in `mms'
from tracker.rb:32:in `run'
```

**Comment by FanaHOVA** [comment]
@margOnline Hey Margo! Can you post the rest of your Tracker file too?

**Comment by margOnline** [comment]
https://github.com/margOnline/fpl-api/blob/master/notification.rb

I had to use a static file as the draftkings url was redirecting. If I remove https://github.com/margOnline/fpl-api/blob/master/notification.rb#L24 and replace the text body with a string that joins the players array, everything works fine. I receive the text.

Great tutorial, btw, I'm modifying it to eventually use for the UK premier league.

**Comment by FanaHOVA** [comment]
@margOnline if you change the URL you wanted to use from `draftkings.com` to `draftkings.co.uk` it should work! what version of Ruby are you using? Looks like the error is related to the imgur gem, not your code, so maybe you are using an outdated version? I'm gonna try and run it locally when I get back after lunch. 

Would love to help you figure it, the PL is probably my favorite football league right now, here in Italy things got pretty boring since Juventus stepped it up big time :P

**Comment by margOnline** [comment]
The .co.uk version of the url worked, thanks. Yea, I think there is an issue with the imgur gem. I downloaded the source code and started playing around with it but now the day job is calling. I'll try to look at it more at the weekend and find a premier league source. Definitely the best league... imo

**Comment by FanaHOVA** [comment]
@margOnline Have fun! Feel free to post here again whenever you move forward with it.

**Comment by margOnline** [comment]
Hi Alessio, I've started on the updates to use the fantasy premier league api. You can view it at https://github.com/margOnline/fpl-api/tree/dreamteam. 

I've done the db setup a bit differently as I only want to run it once. It works fine, creates the image, but the twilio api has stopped working. I get an error on https://github.com/margOnline/fpl-api/blob/dreamteam/notification.rb#L21  
```
.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/common.rb:146:in `split': bad URI(absolute but no path): api.twilio.com:// (URI::InvalidURIError)
```
Have you seen this before?  I'm wondering if its a gem dependency issue? The full stack trace is
```
/Users/me/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/common.rb:146:in `split': bad URI(absolute but no path): api.twilio.com:// (URI::InvalidURIError)
  from /Users/me/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/common.rb:211:in `parse'
  from /Users/me/.rvm/rubies/ruby-2.1.5/lib/ruby/2.1.0/uri/common.rb:747:in `parse'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/faraday-0.13.1/lib/faraday/connection.rb:453:in `proxy_from_env'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/faraday-0.13.1/lib/faraday/connection.rb:85:in `initialize'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/faraday-0.13.1/lib/faraday.rb:67:in `new'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/faraday-0.13.1/lib/faraday.rb:67:in `new'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/http/http_client.rb:21:in `_request'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/http/http_client.rb:54:in `request'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/rest/client.rb:70:in `request'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/framework/domain.rb:20:in `request'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/framework/version.rb:51:in `request'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/framework/version.rb:154:in `create'
  from /Users/me/.rvm/gems/ruby-2.1.5/gems/twilio-ruby-5.3.1/lib/twilio-ruby/rest/api/v2010/account/message.rb:71:in `create'
  from /Users/me/ruby_projects/fpl-api/notification.rb:21:in `mms'
  from fpl_tracker.rb:14:in `run'
```

**Comment by FanaHOVA** [comment]
@margOnline Sorry about the delay, I was out for the weekend. Are the ENV variables set properly? Can you try and replace the `ENV['..']` with `ENV.fetch('...')` and see if it raises any errors while running?

**Comment by margOnline** [comment]
The `ENV` vars are correct. I've pulled the repo onto another computer (no code changes) and am able to run it and receive the text. It must be something to do with my computer set up with ruby version or gem dependencies? Its very strange.

The above only works for text and not images. I've realised that I'm probably getting the imgur-api error because twilio does not support sending images outside the US and Canada.
